ratelimit.c

```diff
@@ -20,9 +20,46 @@
  * SOFTWARE.
  */
 
-#include <stdlib.h>
-#include <sys/time.h>
 #include "redismodule.h"
+#include <stdlib.h>
+#include <time.h> // clock_gettime, CLOCK_MONOTONIC and CLOCK_REALTIME
+
+#ifndef CLOCK_REALTIME
+#include <sys/time.h> // gettimeofday
+#endif
+
+/* nanoseconds per second */
+#define NSEC_PER_SEC 1000000000LL
+
+/* microseconds per second */
+#define USEC_PER_SEC 1000000LL
+
+/* miliseconds per second */
+#define MSEC_PER_SEC 1000LL
+
```

Comment:
I think there should be another constant called `USEC_PER_NSEC` just so the code is more readable.

Reply:
The `gettimeofday()` call with the confusing logic is now removed. I don't think the `USEC_PER_NSEC` needs to be added anymore?